File: action-type/15-chain-of-responsibility/java/PoliticalFilter.java
package main;

// 政治敏感词过滤器
public class PoliticalFilter implements SensitiveFilter {
    
    @Override
    public String doFilter(String text, FilterChain chain) {
    
        String result = text.replaceAll("国家领导人", "[敏感词]");
        System.out.println("[PoliticalFilter] 处理后: " + result);
    
        return chain.doFilter(result);
    }
}

File: action-type/15-chain-of-responsibility/go/political_filter.go
package main

import (
	"fmt"
	"regexp"
)

// PoliticalFilter 政治敏感词过滤器

type PoliticalFilter struct{}

func (f *PoliticalFilter) Filter(text string, chain *FilterChain) string {

	result := text
	result = regexp.MustCompile(`国家领导人`).ReplaceAllString(result, "[敏感词]")
	fmt.Println("[PoliticalFilter] 处理后:", result)

	return chain.Filter(result)
}
